### ISSUE TITLE
Add ARN to AWS Kinesis Data Streams SDK instrumentation

### ISSUE THREAD

**Issue by obenkenobi** [newrelic/newrelic-java-agent#2017]
Requires the [account ID solution for AWS SDKs](https://github.com/newrelic/newrelic-java-agent/issues/2003) is done and AWS Kinesis Datastreams SDK is [instrumented](https://github.com/newrelic/newrelic-java-agent/issues/2016)

**Comment by obenkenobi** [comment]
The functionality on the agent side is essentially "code complete" and merged into the main branch of the agent but not done.
Waiting on end to end validation to be completed till the ticket is closed.

**Comment by obenkenobi** [comment]
End to end validation works
